### PR TITLE
refactor: Rename `SolutionData` -> `Solution`, `Solution` -> `SolutionSet`, `decision_variables` -> `predicate_data`

### PR DESCRIPTION
**Pull request by mitchmindtree**

### CODE REVIEW

**Review by mohammadfawaz — APPROVED** [review]
Makes sense! Thanks!

File: crates/check/src/solution.rs [hunk 1]
```diff
@@ -40,12 +40,11 @@ pub struct CheckPredicateConfig {
...
 }
 
-/// Required impl for retrieving access to any [`SolutionData`]'s [`Predicate`]s during check.
+/// Required impl for retrieving access to any [`Solution`]'s [`Predicate`]s during check.
 pub trait GetPredicate {
     /// Provides immediate access to the predicate with the given content address.
     ///
-    /// This is called by [`check_predicates`] for each predicate in each solution data being
-    /// checked.
+    /// This is called by [`check_predicates`] for each predicate in each solution being checked.
```
Comment (mohammadfawaz) [inline crates/check/src/solution.rs:47]: `check_predicates` -> `check_set_predicates` (in a few places in this file)

File: crates/check/src/lib.rs [hunk 2]
```diff
@@ -13,14 +13,14 @@
 //!
 //! ## Solution Validation
 //!
-//! - [`solution::check`] validates an unsigned solution.
-//! - [`solution::check_data`] validates a solution's data slice.
-//! - [`solution::check_state_mutations`] validates a solution's state mutation slice.
+//! - [`solution::check_set`] validates a solution set.
```
Comment (nfurfaro) [inline crates/check/src/lib.rs:16]: Is it relevant whether the solution set is signed or not here?

**Review by nfurfaro — APPROVED** [review]
LGTM.

**Review by nfurfaro — APPROVED** [review]

File: crates/check/src/lib.rs (hunk 2 above)
Comment (mitchmindtree) [reply crates/check/src/lib.rs:16]: Not any longer, we used to have signed solutions long ago but there's no longer any signing of solutions or contracts in-protocol

**Review by mohammadfawaz — APPROVED** [review]

**Pull request merged**